src/components/Settings/TextGenerationSettings.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect, ReactNode } from "react";
+import Slider from "@mui/material/Slider";
+import { LLMGenerationParameters } from "electron/main/Store/storeConfig";
+import { Button } from "@material-tailwind/react";
+
+interface TextGenerationSettingsProps {
+  children?: ReactNode; // Define children prop
+}
+const TextGenerationSettings: React.FC<TextGenerationSettingsProps> = ({
+  children,
+}) => {
+  const [textGenerationParams, setTextGenerationParams] =
+    useState<LLMGenerationParameters>({
+      temperature: 0.7, // Default temperature value
+      maxTokens: 2048, // Default maxTokens value
+      // Include other default values as necessary
+    });
+
+  const [userHasMadeUpdate, setUserHasMadeUpdate] = useState(false);
+  // const [temperature, setTemperature] = useState<number | null>();
+  // const [maxTokens, setMaxTokens] = useState<number | null>();
+
+  useEffect(() => {
+    const params = window.electronStore.getLLMGenerationParams();
+    if (params) {
+      setTextGenerationParams(params);
+    }
+  }, []);
+
+  const handleSave = () => {
+    // Execute the save function here
+    if (textGenerationParams) {
+      window.electronStore.setLLMGenerationParams(textGenerationParams);
+      setUserHasMadeUpdate(false);
+    }
+  };
+
+  return (
+    <div className="w-full bg-gray-800 rounded pb-7">
+      {children}
+      <h2 className="text-2xl font-semibold mb-0 text-white">
+        Text Generation
+      </h2>{" "}
+      <p className="mt-2 text-sm text-gray-100 mb-1">Temperature:</p>
+      <Slider
+        aria-label="Temperature"
+        value={textGenerationParams.temperature}
+        valueLabelDisplay="auto"
+        step={0.1}
+        marks
+        min={0}
+        max={2}
+        onChange={(event, val) => {
+          setUserHasMadeUpdate(true);
+          const newTemperature = Array.isArray(val) ? val[0] : val;
+          setTextGenerationParams({
+            ...textGenerationParams,
+            temperature: newTemperature,
+          });
+        }}
+      />
```

Comment:
The preview exceeds the menu a little, it looks weirdly unjustified. Perhaps we can add some padding to this component?
![image](https://github.com/reorproject/reor/assets/58682208/4765e5ec-4f9c-4056-a252-731a37795de0)